src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,39 @@
+package duke.command;
+
+import java.util.ArrayList;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.Ui;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+public class FindCommand extends Command {
+    private String InputToSearch;
+    public FindCommand(String input) {
+        this.InputToSearch = input;
+    }
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        ArrayList<Task> foundTasks = taskList.findTasks(this.InputToSearch);
+        if (foundTasks.size() == 0) {
+            ui.showMessage("We could not find any matching tasks.");
+        } else {
+            ui.showMessage("Here are the matching tasks in your list: ");
+            for (Task foundTask : foundTasks) {
+                if (foundTask instanceof Todo) {
+                    ui.showMessage(foundTasks.indexOf(foundTask) + 1 + ". [" + foundTask.getSymbol() + "] "
+                            + "[" + foundTask.getStatusIcon() + "] " + foundTask.getDescription());
+                } else {
+                    ui.showMessage(foundTasks.indexOf(foundTask) + 1 + ". [" + foundTask.getSymbol() + "] "
+                                  + "[" + foundTask.getStatusIcon() + "] " + foundTask.getDescription()
+                                  + " (" + foundTask.getDuedateString() + ")");
+                }
+            }
+        }
+    }
```

Comment:
Seems like an arrow-head style code, maybe perhaps the method is too long and too deeply nested. Maybe try abstracting some of the code into another method?